Solutions/Branwen/GUI.cs

```diff
@@ -9,324 +9,460 @@
+			string insertPathsStatement = "INSERT INTO SaurutobiMediaPaths (FileName, PartOfPath, Path, MediaDrive) VALUES ";
+
+			bool firstFileVal = true;
+			foreach (BranwenFileInfo file in files)
+			{
+				if (!firstFileVal)
+				{
+					insertFilesStatement += ", ";
+					insertPathsStatement += ", ";
+				}
+				else
+				{
+					firstFileVal = false;
+				}
+
+				//File Name
+				insertFilesStatement += "('" + file.Name.Replace("'", "''") + "', ";
+
+				//File Extension
+				insertFilesStatement += "'" + file.Extension + "', ";
+
+				//File Size
+				insertFilesStatement += file.FileSize + ", ";
+
+				//Media Type
+				insertFilesStatement += "'" + file.Path[1].Replace("'", "''") + "', ";
+
+				//MediaDrive
+				insertFilesStatement += file.MediaDrive + ")";
+
+				bool firstPathVal = true;
+				for (int i = 0; i < file.Path.Count; i++)
+				{
+					if (!firstPathVal)
+					{
+						insertPathsStatement += ", ";
+					}
+					else
+					{
+						firstPathVal = false;
+					}
+					//File Name
+					insertPathsStatement += "('" + file.Name.Replace("'", "''") + "', ";
+
+					//PartOfPath
+					insertPathsStatement += i + ", ";
+
+					//Path
+					insertPathsStatement += "'" + file.Path[i].Replace("'", "''") + "', ";
+
+					//MediaDrive
+					insertPathsStatement += file.MediaDrive + ")";
+				}
+
+				fileCount++;
+			}
+			insertFilesStatement += ";";
+			insertPathsStatement += ";";
+
+			MySqlCommand insertCommand = new MySqlCommand(insertFilesStatement, mySqlConnection);
+			insertCommand.ExecuteNonQuery();
+			insertCommand = new MySqlCommand(insertPathsStatement, mySqlConnection);
+			insertCommand.ExecuteNonQuery();
+		}
+
+		/// <summary>
+		/// Writes an Excel document with all the contentes of the database
+		/// </summary>
+		/// <param name="mySqlConnection"></param>
+		/// <param name="outputFile"></param>
+		private void ExportDatabaseToFile(MySqlConnection mySqlConnection, string outputFile)
+		{
+			//Get all the MediaTypes
+			List<string> mediaTypes = new List<string>();
+			MySqlCommand command = new MySqlCommand("SELECT MediaType FROM SaurutobiMediaFiles GROUP BY MediaType;", mySqlConnection);
+			using (MySqlDataReader mediaTypeReader = command.ExecuteReader())
+			{
+				while (mediaTypeReader.Read())
+				{
+					mediaTypes.Add(mediaTypeReader["MediaType"].ToString());
+				}
+				mediaTypeReader.Close();
+			}
+
+			//Make sure the outputFile doesn't exist
+			if (File.Exists(outputFile))
+			{
+				File.Delete(outputFile);
+			}
+
+			//Create worksheet
+			SpreadsheetDocument spreadsheetDocument = SpreadsheetDocument.Create(outputFile, DocumentFormat.OpenXml.SpreadsheetDocumentType.Workbook);
+			WorkbookPart workbookpart = spreadsheetDocument.AddWorkbookPart();
+			workbookpart.Workbook = new Workbook();
+			Sheets sheets = spreadsheetDocument.WorkbookPart.Workbook.AppendChild(new Sheets());

```

Comment:
yea, but that's some more work that I didn't warrant as required.

On Thu, Dec 17, 2015 at 11:27 PM, Alex Ayerdi <notifications@github.com>
wrote:

> In Solutions/Branwen/GUI.cs
> <https://github.com/LbKStudios/Branwen/pull/5#discussion_r48000471>:
>
> > +					mediaTypes.Add(mediaTypeReader["MediaType"].ToString());
> > +				}
> > +				mediaTypeReader.Close();
> > +			}
> > +
> > +			//Make sure the outputFile doesn't exist
> > +			if (File.Exists(outputFile))
> > +			{
> > +				File.Delete(outputFile);
> > +			}
> > +
> > +			//Create worksheet
> > +			SpreadsheetDocument spreadsheetDocument = SpreadsheetDocument.Create(outputFile, DocumentFormat.OpenXml.SpreadsheetDocumentType.Workbook);
> > +			WorkbookPart workbookpart = spreadsheetDocument.AddWorkbookPart();
> > +			workbookpart.Workbook = new Workbook();
> > +			Sheets sheets = spreadsheetDocument.WorkbookPart.Workbook.AppendChild(new Sheets());
>
> Personally I don't like looking at all of this DocumentFormat.OpenXml API
> shit. I'd create an object that handles all of this for you behind the
> scenes so you can just pass it a few parameters to get it started and you
> can work with your excel documents easier. Also, you can name your members
> more intuitive things instead of having to deal with the "worksheetpart"
> shit.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LbKStudios/Branwen/pull/5/files#r48000471>.
>